.github/workflows/publish-docs.yml

```diff
@@ -0,0 +1,36 @@
+name: Publish Docs
+
+on:
+  push:
+    branches: [ master ]
```

Comment:
We also need versioned branches to be included.

https://github.com/StackStorm/st2docs/blob/6c65852f42ea6a687c1b87e25ad67a320baa1538/.circleci/config.yml#L72-L75

Once merging this PR, the same change should be pushed to all the versioned branches (or at least the last 5-10) to get them generated.
https://github.com/StackStorm/st2docs/branches

Reply:
full regex filtering does not seem to be supported with GHA: https://github.community/t/using-regex-for-filtering/16427

subset filtering is supported: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

perhaps we could do something like:
```
on:
  push:
    branches:
      - master
      - v[0-9]+.[0-9]+
```

it'd need to be tested, but I think we have another problem with subdirectories (see comment below)

Reply:
confirmed the above works.  will resolve this thread when I'm done testing

Reply:
https://github.com/StackStorm/st2docs/pull/1129/files#diff-7b112bd01a3c7000d73727a7e66cbfb9cf00a47702c94fda157e23f46e20db17R17